app/controllers/api/v2/json_web_tokens_controller.rb

```diff
@@ -0,0 +1,48 @@
+module Api
+  module V2
+    class JsonWebTokensController < V2::BaseController
+      include Api::Version2
+
+      before_action :find_user
+
+      api :POST, "/users/:id/json_web_tokens", N_("Generate JSON web token")
+      param :id, String, required: true
+      param :expires_at, DateTime, desc: N_("Expiration date of the token. If not set, token will never expire")
+      def create
+        render json: { json_web_token: @user.jwt_token!(expiration: expires_at) }
+      end
+
+      api :DELETE, "/users/:id/json_web_tokens", N_("Invalidates all user's JSON web tokens")
+      param :id, String, required: true
```

Comment:
maybe worht adding description, that it is a user id?

Also I'd like to reiterate the question: does one user have only one web token? if for any reason in the future we would want more web_tokens per user, this API would require some id of the token that is about to be destroyed and the `:id` should be renamed to `:user_id` and `:id` then should be web_token id by convention.

Do we want to at least have the user id param named `user_id` from the day 1?

Reply:
> maybe worht adding description, that it is a user id?

Added

> does one user have only one web token?

Sorry missed your comment. No, you can have unlimited number of tokens. We do not save tokens to the database, (therefore don't have `id`), so when user call `invalidate` endpoint, we invalidate all of them.

> Do we want to at least have the user id param named user_id from the day 1?

As my comment above, we do not store `JWT` in database, so there is no need for it.